src/helpers.rs

```diff
@@ -320,3 +322,10 @@ pub fn get_rescript_version(bsc_path: &str) -> String {
         .replace('\n', "")
         .replace("ReScript ", "")
 }
+
+pub fn read_file(path: &Path) -> Result<String, std::io::Error> {
```

Comment:
but we need to do this with multiple keywords (extension points / annotations), so reading the file once in memory and scanning for them is faster I think

Reply:
we can only stop looking for them if we found all, so will only be faster in files that have them all in the top of the file, don't think that happens very often

Reply:
Hmm. Yeah, fair enough. That's tricky.
Two ideas:
- Perhaps we can do this on an initial scan and store the PPX flags within the file metadata
- Then only update this for the files that change
- Bonus: Not run every PPX on all files (because I think that for a PPX to figure out if it needs running or not, it will need to do an AST map - so read + map + ?write)

Reply:
Here's some inspiration from the Relay compiler that extracts GraphQL and a custom "RelayResolver" doc block syntax: https://github.com/facebook/relay/blob/main/compiler/crates/extract-graphql/src/lib.rs

It tries to ignore things in comments IIRC, and it tracks cols and lines, which would be useful later on if we want to continue exploring https://github.com/teamwalnut/rewatch/issues/62.

I'm not familiar in depth with `str.contains`, maybe running it once for each extension point has negligible overhead, but I think it'd be ideal to do a single pass over the contents and just extract all present extension points (again, optionally including the content and pos later on then if we want to explore the embedded lang thing I linked to). The full file will need to be looped through anyway unless all possible extension points are present high up in the same file, which sounds like something that won't happen often.

@rolandpeelen I believe this PR already only applies the PPXes when appropriate (as in - when the extension points are available).

Reply:
It's already a big improvement. Let's do this for the current community ppx's, happy to add more like the relay one. The scanning in memory isn't a big overhead, but we can look into optimizations to only do a single pass over the data as an optimization later.

Reply:
I think we can extract all extension points and annotations in a file with a simple parse step, and then use that to selectively run the ppxs.